File: services/order-service/internal/domain/menu.go
package domain

// MenuCategory represents a category of menu items (e.g., Закуски, Горячее).
type MenuCategory struct {
	ID        string
	Name      string
	SortOrder int32
}

// MenuItem represents a single dish in the menu catalog.
type MenuItem struct {
	ID         string
	CategoryID string
	Name       string
	Price      float64
	ImageURL   string
	Available  bool
}

File: frontend/src/lib/schemas/menu.ts
import { z } from 'zod';

export const MenuCategorySchema = z.object({
  id: z.string(),
  name: z.string(),
  sortOrder: z.number().int(),
});

export const MenuItemSchema = z.object({
  id: z.string(),
  categoryId: z.string(),
  name: z.string(),
  price: z.number().positive(),
  imageUrl: z.string(),
  available: z.boolean(),
});

export type MenuCategoryDTO = z.infer<typeof MenuCategorySchema>;
export type MenuItemDTO = z.infer<typeof MenuItemSchema>;
